### PR TITLE
feat: adds terraform plan action

### PR DESCRIPTION
**Pull request by arein**
See https://github.com/WalletConnect/ts-relay/pull/141 for context

### CODE REVIEW

File: actions/plan-terraform/action.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+name: 'Plan Infra'
+description: 'Plan Terraform infrastructure'
+inputs:
+  environment:
+    description: 'The environment to deploy to'
+    required: true
+  aws-access-key-id:
```
[the remark this answers is not in the document]
Comment (arein) [reply actions/plan-terraform/action.yml:7]: We already use composite actions with access/secret keys. I'm open to changing our AWS credential retrieval and to switching to roles but that should be a dedicated task and not part of this distinct commit.

Further questions that need to be addressed for moving to roles:
+ enable us to use GitHub's OIDC provider
+ creation of app specific deployment roles

File: actions/plan-terraform/action.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        GRAFANA_AUTH: ${{ inputs.grafana-api-key }}
+      run: terraform -chdir=terraform workspace select ${{ inputs.environment }}
+    - name: Terraform Plan
+      id: plan
+      shell: bash
+      env:
+        GRAFANA_AUTH: ${{ inputs.grafana-api-key }}
+      run: |
+        terraform -chdir=terraform plan -var-file="vars/${{ inputs.environment }}.tfvars" -no-color -out=/tmp/plan.tfplan >> /tmp/plan.txt
+        cat /tmp/plan.txt
```
[the remark this answers is not in the document]
Comment (arein) [reply actions/plan-terraform/action.yml:58]: what part in particular? but yes ([sample run](https://github.com/WalletConnect/ts-relay/runs/7101754920?check_suite_focus=true))

[the remark this answers is not in the document]
Comment (ffrizzo) [reply actions/plan-terraform/action.yml:65]: the `cat` on plan.txt

File: actions/plan-terraform/action.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ffrizzo) [reply actions/plan-terraform/action.yml:7]: Can you create a task for this, please?

[the remark this answers is not in the document]
Comment (arein) [reply actions/plan-terraform/action.yml:7]: https://github.com/orgs/WalletConnect/projects/9/views/4

**Review by ffrizzo — APPROVED** [review]

**Review by HarryET — APPROVED** [review]

**Pull request merged**